beetmoverscript/templates/mini_artifact_manifest.yml

```diff
@@ -0,0 +1,603 @@
+metadata:
+  description: Maps {{ product }} artifacts to pretty names
+  name: Beet Mover Manifest
+  owner: release@mozilla.com
+s3_bucket_paths:
+  devedition_candidates: pub/devedition/candidates/{{ version }}-candidates/build{{ build_number }}/
+  devedition_candidates_repacks: pub/devedition/candidates/{{ version }}-candidates/build{{ build_number }}/
+  fennec_candidates_repacks: pub/mobile/candidates/{{ version }}-candidates/build{{ build_number }}/
+  fennec_nightly: pub/mobile/nightly/
+  fennec_nightly_repacks: pub/mobile/nightly/
+  firefox_candidates: pub/firefox/candidates/{{ version }}-candidates/build{{ build_number }}/
+  firefox_candidates_repacks: pub/firefox/candidates/{{ version }}-candidates/build{{ build_number }}/
+  firefox_nightly: pub/firefox/nightly/
```

Comment:
I think splitting these bucket paths out is smart in that it avoids repeating these strings throughout the file. I'm wondering if we also want to put in things like nightly dated dir paths here, or in an `s3_subdirs` section.

Reply:
Hm, I think `s3_subdirs` is a good idea. Especially because having `pretty_path`s (hence a list instead of a value) for some of the entries might be confusing even more and look ugly. 

If we were to choose `target.crashreporter-symbols.zip:` (e.g. from [here](https://github.com/mozilla-releng/beetmoverscript/blob/master/beetmoverscript/templates/firefox_nightly.yml#L62), the way I'd see the manifest for it would be something like: 
```
    target.crashreporter-symbols.zip:
      default: target.crashreporter-symbols.zip
      devedition:
        candidates:
          pretty_name: firefox-{{ version }}.crashreporter-symbols.zip
          pretty_path: {{ platform }}/{{ locale }}/firefox-{{ version }}.crashreporter-symbols.zip
...
        nightly:
          pretty_name: firefox-{{ version }}.{{ locale }}.{{ platform }}.crashreporter-symbols-full.zip
          pretty_path: ''
          s3_subdirs: 
               - latest-{{ branch }}/
               - {{ upload_date }}-{{ branch }}/
```

`pretty_path` would stay empty in this case. 
I see full path of an artifact being constructed like:

http://archive.mozilla.org/ + `s3_bucket_paths` + `s3_subdirs`(if any) + `pretty_path` + `pretty_name`

This assumes that the `pretty_name` +`pretty_path` stay the same if we have `s3_subdirs` which I'm not 100% sure is the case for all artifacts. I can't think a counter-example but I'm wondering if that's possible.